doc/operations/configuration.rst

```diff
@@ -0,0 +1,235 @@
+    - ``recovery_threshold``. Note that if the recovery threshold is set to ``0``, it is automatically set to the number of recovery members specified in ``members``.
+    - ``reconfiguration_type``. The type of reconfiguration for new nodes. Default value: ``OneTransaction``.
+
+Example:
+
+.. code-block:: json
+
+    "start": {
+        "constitution_files": ["actions.js", "validate.js", "resolve.js", "apply.js"],
+        "members": [
+            {"certificate_file": "member0_cert.pem", "data_json_file": null, "encryption_public_key_file": "member0_enc_pubk.pem"},
+            {"certificate_file": "member1_cert.pem", "data_json_file": "member1_data.json", "encryption_public_key_file": null},
+            {"certificate_file": "member2_cert.pem", "data_json_file": null, "encryption_public_key_file": "member2_enc_pubk.pem"}
+        ],
+        "service_configuration":
+        {
+            "recovery_threshold": 0,
+            "max_allowed_node_cert_validity_days": 365,
+            "reconfiguration_type": "OneTransaction"
+        }
+    }
+
+.. _join configuration:
+
+``join``
+~~~~~~~~
+
+.. note:: This only needs to be set when the node is started in ``join`` mode.
+
+- ``target_rpc_address``: Address (hostname and port) of a node of the existing service to join.
+- ``join_timer_ms``: Interval (ms) at which the node sends join requests to the existing network. Default value: ``1000`` ms.
+
+Example:
+
+.. code-block:: json
+
+    "join": {
+        "join_timer_ms": 1000,
+        "target_rpc_address": {"hostname": "127.0.0.1", "port": "8080"}
+    }
+
+``ledger``
+~~~~~~~~~~
+
+- ``ledger_dir``: Path to main ledger directory. Default value: ``ledger``.
+- ``read_only_ledger_dirs``: Optional. Paths to read-only ledger directories. Note that only ``.committed`` files will be read from these directories. Default value: ``[]``.
+- ``ledger_chunk_bytes``: Minimum size (bytes) of the current ledger file after which a new ledger file (chunk) is created. Default value: ``5000000`` bytes.
+
+``snapshots``
+~~~~~~~~~~~~~
+
+- ``snapshots_dir``: Path to snapshot directory. Default value: ``snapshots``.
+- ``snapshot_tx_interval``: Minimum number of transactions between snapshots. Default value: ``10000``.
+
+``logging``
+~~~~~~~~~~~
+
+- ``host_log_level``: Logging level for the `untrusted host`. Default value: ``INFO``.
+
+.. note:: While it is possible to set the host log level at startup, it is not possible to change the log level of the enclave.
+
+- ``log_format_json``: If ``true``, node logs will be formatted as JSON. Default value: ``false``.
+
+``consensus``
+~~~~~~~~~~~~~
+
+- ``type``: Type of consensus protocol. Only ``CFT`` (Crash-Fault Tolerant) is currently supported in production. Default value: ``CFT``.
+- ``timeout_ms``: Hearbeat interval (ms) at which primary node sends messages to backup nodes to maintain primary-ship. This should be set to a significantly lower value than ``election_timeout_ms``. Default value: ``100`` ms.
```

Comment:
Heartbeats aren't quite the whole story, it's also how often the primary retries sending messages that haven't been acked.